### DISCUSSION TITLE
Reason for fidelity equivalence in entanglement generation

### DISCUSSION THREAD

**Discussion by ghost** [sequence-toolbox/SeQUeNCe#68] [Q&A]
Hello,

I am wondering why the assertion here https://github.com/sequence-toolbox/SeQUeNCe/blob/8cbcbd31f013e511d5cbaf96399711161ddadabe/src/entanglement_management/generation.py#L142 was added. Can you explain, @xwu64, the purpose of ensuring fidelity equivalence here?

**Comment by xwu64** [comment]
I don't think it's necessary if the generation is implemented on the real testbed. At here, we are not clear what will the fidelity of entanglement from the generation protocol. Therefore, we just assume both sides of protocols have a consensus of the fidelity of entanglement after the generation protocol. To avoid inconsistency of fidelity of entangled memories, we add this assertion.

Please let us know if you have ideas about this implementation.

**Reply by ghost** [reply]
Gotcha, thanks!
